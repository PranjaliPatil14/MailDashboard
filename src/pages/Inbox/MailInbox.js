import React, { useContext, useEffect, useState } from "react";
import Header from "../../components/header/Header";
import "./mailInbox.css";
import Menu from "./menu/Menu";
import Inbox from "./mail/Inbox";
import { UserContext } from "../../context/userContext";
import ComposeEmail from "../ComposeEmail/ComposeEmail";

const MailInbox = () => {
  const { setLoggedInUser } = useContext(UserContext);
  const [showCompose, setShowCompose] = useState(false);

  useEffect(() => {
    setLoggedInUser({
      id: 2,
      name: "abc",
      role: "Art Director",
      profileImage: "profile-image.png",
    });
  }, [setLoggedInUser]);
  return (
    <div className="page">
      {showCompose && <ComposeEmail setShowCompose={setShowCompose} />}
      <Header />
      <div className="page-content">
        <Menu setShowCompose={setShowCompose} />
        <Inbox />
      </div>
    </div>
  );
};

export default MailInbox;
