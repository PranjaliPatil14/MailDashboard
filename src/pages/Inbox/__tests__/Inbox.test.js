import React from "react";
import { render } from "@testing-library/react";
import MailInbox from "../MailInbox";

describe("MailInbox", () => {
  test("should render inbox", () => {
    const { asFragment } = render(<MailInbox />);

    expect(asFragment()).toMatchSnapshot();
  });
});
