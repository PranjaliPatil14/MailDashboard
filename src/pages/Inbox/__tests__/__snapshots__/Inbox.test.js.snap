// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inbox should render inbox 1`] = `
<DocumentFragment>
  <div
    class="page"
  >
    <header
      class="app-header"
    >
      <div
        class="header-left"
      >
        <svg
          class="header-hamburger"
          fill="none"
          height="18"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="3"
            x2="21"
            y1="12"
            y2="12"
          />
          <line
            x1="3"
            x2="21"
            y1="6"
            y2="6"
          />
          <line
            x1="3"
            x2="21"
            y1="18"
            y2="18"
          />
        </svg>
        <input
          class="header-search"
          placeholder="Search for Something..."
          type="search"
        />
      </div>
      <div
        class="header-right"
      >
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
          />
          <polyline
            points="22,6 12,13 2,6"
          />
        </svg>
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 8A6 6 0 0 0 6 8c0 7-3 9-3 9h18s-3-2-3-9"
          />
          <path
            d="M13.73 21a2 2 0 0 1-3.46 0"
          />
        </svg>
        <button
          class="header-logout"
          type="submit"
        >
          <svg
            fill="none"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
            />
            <polyline
              points="7 10 12 15 17 10"
            />
            <line
              x1="12"
              x2="12"
              y1="15"
              y2="3"
            />
          </svg>
          <span>
            Logout
          </span>
        </button>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
